tests/handlers/test_profile.py

```diff
@@ -147,3 +161,20 @@ def test_set_my_avatar(self):
             (yield self.store.get_profile_avatar_url(self.frank.localpart)),
             "http://my.server/pic.gif",
         )
+
+    @defer.inlineCallbacks
+    def test_set_my_avatar_if_disabled(self):
+        self.hs.config.enable_set_avatar_url = False
+
+        # Set first time avatar is allowed, if avatar is null
+        yield self.store.set_profile_avatar_url(
```

Comment:
I don't think you're testing that setting a profile picture/displayname initially works when this setting is activated.

Reply:
I have updated the tests. It is better?

Reply:
Now that I'm looking at this again, I think the fact that you yield'd successfully did test that it was allowed, but checking that the value was actually stored in the database definitely doesn't hurt :)